pkg/pulumiyaml/run.go

```diff
@@ -935,6 +935,9 @@ func (e *programEvaluator) registerConfig(intm configNode) (interface{}, bool) {
 			markSecret = true
 		}
 	case configNodeProp:
+		if intm.v.IsSecret() {
```

Comment:
Ah this is expecting an Output value, not a PropertyValue. Good ol' `interface{}` failing to catch any errors again.
Shouldn't this just match the code below though pretty much of?
```
v := intm.value()
if intm.v.IsSecret() {
	v = pulumi.ToSecret(v)
}
return v, true
```

Reply:
What you have here doesn't quite work, because `intm.value()` already unwraps the property value (it essentially returns `intm.v.V`), and from my testing we end up with the same result then.  using `pulumi.ToSecret(intm.v)` does work, so we can go with that.